src/components/Documentation/Markdown/index.js

```diff
@@ -1,279 +1,145 @@
-import React from 'react'
+import React, { useCallback, useEffect, useRef } from 'react'
+import rehypeReact from 'rehype-react'
 import PropTypes from 'prop-types'
-import ReactMarkdown from 'react-markdown'
 import Collapsible from 'react-collapsible'
-import kebabCase from 'lodash.kebabcase'
 
-import Router from 'next/router'
+import 'github-markdown-css/github-markdown.css'
+
+import { navigate } from '@reach/router'
 
 import LocalLink from '../../LocalLink'
 import Tooltip from '../../Tooltip'
 
 import Tutorials from '../Tutorials'
 
-import { Light as SyntaxHighlighter } from 'react-syntax-highlighter'
-import { docco } from 'react-syntax-highlighter/dist/cjs/styles/hljs'
-import python from 'react-syntax-highlighter/dist/cjs/languages/hljs/python'
-import yaml from 'react-syntax-highlighter/dist/cjs/languages/hljs/yaml'
-import ini from 'react-syntax-highlighter/dist/cjs/languages/hljs/ini'
-import bash from 'react-syntax-highlighter/dist/cjs/languages/hljs/bash'
-import diff from 'react-syntax-highlighter/dist/cjs/languages/hljs/diff'
-import vim from 'react-syntax-highlighter/dist/cjs/languages/hljs/vim'
-import usage from './lang/usage'
-import dvc from './lang/dvc'
-import linker from './utils/remark-linker'
-
-import { PAGE_DOC } from '../../../consts'
-
 import {
   Button,
   Content,
-  ExternalLink,
   GithubLink,
   NavigationButtons,
   TutorialsWrapper
 } from './styles'
 
-SyntaxHighlighter.registerLanguage('dvc', dvc)
-SyntaxHighlighter.registerLanguage('python', python)
-SyntaxHighlighter.registerLanguage('usage', usage)
-SyntaxHighlighter.registerLanguage('yaml', yaml)
-SyntaxHighlighter.registerLanguage('ini', ini)
-SyntaxHighlighter.registerLanguage('bash', bash)
-SyntaxHighlighter.registerLanguage('vim', vim)
-SyntaxHighlighter.registerLanguage('diff', diff)
-
-function flatten(text, child) {
-  return typeof child === 'string'
-    ? text + child
-    : React.Children.toArray(child.props.children).reduce(flatten, text)
-}
-
-const SLUG_REGEXP = /\s+{#([a-z0-9-]*[a-z0-9]+)}\s*$/
-
-function isTitleHasSlug(title) {
-  return typeof title === 'string' && SLUG_REGEXP.test(title)
-}
-
-export function extractSlugFromTitle(title) {
-  // extracts expressions like {#too-many-files} from the end of a title
-  const meta = title.match(SLUG_REGEXP)
-
-  if (meta) {
-    return [title.substring(0, meta.index), meta[1]]
+const isInsideCodeBlock = elem => {
```

Comment:
You might want to have a look at [closest](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest).

Reply:
Just copied block for now too. Probably we need to update it separately.